server/src/test/java/io/crate/execution/engine/collect/count/InternalCountOperationTest.java

```diff
@@ -84,4 +95,29 @@ public void testCount() throws Exception {
             assertThat(count.get(5, TimeUnit.SECONDS), is(1L));
         }
     }
+
+    @Test
+    public void test_handles_recovering_shard_state_for_partitioned_tables() throws Exception {
```

Comment:
Does it make sense (and is it easy for this test) to have some other shard that is started, so that the count returns value only for this shard + 0 for the recovering one?

Reply:
It would be possible but not sure what that would add? We've plenty of tests covering the normal operation of `count(*)` and this test case is specific for the `syncCount` method

Reply:
Nothing essential, just for a more complex scenario, it's fine as is.